Builder/SolariumSelectQueryBuilder.php

```diff
@@ -211,6 +212,28 @@ public function buildSolariumQueryFromGeneric(ResolvedSelectQueryInterface $quer
             $solariumSpellcheck->setDictionary($query->getSpellcheck()->getDictionary());
         }
 
+        // if grouping on a field then group - and apply grouping sort if applicable
+        if ($query->getGroupingField() !== null) {
+            $groupComponent = $solariumQuery->getGrouping();
+            $groupComponent->addField($query->getGroupingField());
+            $groupComponent->setLimit(1000);
+            $groupComponent->setMainResult(false);
+            $groupComponent->setNumberOfGroups(true);
+
+            $groupingSortCollection = $query->getGroupingSortCollection();
+            if ($groupingSortCollection instanceof SortCollectionInterface) {
+                $sortStringComponents = [];
+                foreach($groupingSortCollection as $groupingSort) {
+                    $sortStringComponents[] = sprintf(
+                        '%s %s',
+                        $groupingSort->getFilter()->getSearchKey(),
+                        ($groupingSort->isDescending()) ? SolariumQuery::SORT_DESC : SolariumQuery::SORT_ASC
+                    );
+                }
+                $groupComponent->setSort(implode(',', $sortStringComponents));
+            }
```

Comment:
this sets the sort order within the group (rather than between groups - which is what the regular sort order does.